src/AccessTokenHandler.ts

```diff
@@ -0,0 +1,43 @@
+import { APIGatewayProxyEvent } from "aws-lambda";
+import { LambdaInterface } from "@aws-lambda-powertools/commons";
+import { Metrics } from "@aws-lambda-powertools/metrics";
+import { Logger } from "@aws-lambda-powertools/logger";
+import { Constants } from "./utils/Constants";
+import { Response } from "./utils/Response";
+import { MessageCodes } from "./models/enums/MessageCodes";
+import { HttpCodesEnum } from "./models/enums/HttpCodesEnum";
+import { AccessTokenRequestProcessor } from "./services/AccessTokenRequestProcessor";
+import { AppError } from "./utils/AppError";
+
+const { POWERTOOLS_METRICS_NAMESPACE = Constants.BAV_METRICS_NAMESPACE, POWERTOOLS_LOG_LEVEL = "DEBUG", POWERTOOLS_SERVICE_NAME = Constants.ACCESSTOKEN_LOGGER_SVC_NAME } = process.env;
+
+const logger = new Logger({
+	logLevel: POWERTOOLS_LOG_LEVEL,
+	serviceName: POWERTOOLS_SERVICE_NAME,
+});
+
+const metrics = new Metrics({ namespace: POWERTOOLS_METRICS_NAMESPACE, serviceName: POWERTOOLS_SERVICE_NAME });
+
+export class AccessToken implements LambdaInterface {
+
+	@metrics.logMetrics({ throwOnEmptyMetrics: false, captureColdStartMetric: true })
+	async handler(event: APIGatewayProxyEvent, context: any): Promise<Response> {
+
+		// clear PersistentLogAttributes set by any previous invocation, and add lambda context for this invocation
+		logger.setPersistentLogAttributes({});
+		logger.addContext(context);
+
+		try {
+			logger.info("Starting AccessTokenRequestProcessor");
+			return await AccessTokenRequestProcessor.getInstance(logger, metrics).processRequest(event);
+		} catch (error: any) {
+			logger.error({ message: "An error has occurred.", error, messageCode: MessageCodes.SERVER_ERROR });
+			if (error instanceof AppError) {
+				return new Response(error.statusCode, error.message);
+			}
+			return new Response(HttpCodesEnum.SERVER_ERROR, "Server Error");
```

Comment:
Would you mind adding the error codes to the [runbook](https://govukverify.atlassian.net/wiki/spaces/FTFCRI/pages/3786211381/BAV+-+POST+token+Lambda) please?

Reply:
Updated.